rita_client/src/operator_update/mod.rs

```diff
@@ -680,134 +649,65 @@ fn contains_forbidden_key(map: Map<String, Value>, forbidden_values: &[&str]) ->
-        ];
-        let key_file: &str = "auth_remove_keys";
-
-        let operator_key = touch_temp_file(key_file);
-
-        let _update = update_authorized_keys(added_keys, removed_keys, key_file);
-        let result = parse_keys(key_file);
-        assert!(result.contains(&operator_key.to_string()));
-
-        assert_eq!(result.len(), 1);
-
-        remove_temp_file(key_file).unwrap();
-    }
-    #[test]
-    fn test_removing_existing_key() {
-        let added_keys = vec![];
-        let key_file: &str = "remove_keys";
-
-        let operator_key = touch_temp_file(key_file);
-        let removed_keys = vec![String::from(operator_key)];
-        let _update = update_authorized_keys(added_keys, removed_keys.clone(), key_file);
-
-        let result = parse_keys(key_file);
-        for item in result {
-            assert_eq!(item, removed_keys[0].to_string());
-        }
+    res
+}
 
-        remove_temp_file(key_file).unwrap();
-    }
-    #[test]
-    fn test_authorized_keys_create_if_missing() {
-        let added_keys = vec![
-            String::from("ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHFgFrnSm9MFS1zpHHvwtfLohjqtsK13NyL41g/zyIhK test@hawk-net ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDVF1POOko4/fTE/SowsURSmd+kAUFDX6VPNqICJjn8eQk8FZ15WsZKfBdrGXLhl2+pxM66VWMUVRQOq84iSRVSVPA3abz0H7JYIGzO8psTweSZfK1jwHfKDGQA1h1aPuspnPrX7dyS1qLZf3YeVUUi+BFsW2gSiMadbS4zal2c2F1AG5Ezr3zcRVA8y3D0bZxScPAEX74AeTFcimHpHFyzDtUsRpf0uSEXZcMFqX5j4ETKlIs28k1v8LlhHo91IQYHEtbyi/I1M0axbF4VCz5JlcbAs9LUEJg8Kx8LxzJSeSJbxVwyk5WiEDwVsCL2MAtaOcJ+/FhxLb0ZEELAHnXFNSqmY8QoHeSdHrGP7FmVCBjRb/AhVUHYvsG94rO3Ij4H5XsbsQbP3AHVKbvf387WB53Wga7VrBXvRC9aDisetdP9+4/seVIBbOIePotaiHoTyS1cJ+Jg0PkKy96enqwMt9T1Wt8jURB+s/A/bDGHkjB3dxomuGxux8dD6UNX54M= test-rsa@hawk-net"),
-        ];
-        let removed_keys: Vec<String> = vec![];
-        let key_file: &str = "create_keys_file";
-        let _update = update_authorized_keys(added_keys, removed_keys, key_file);
-        assert!(Path::new(key_file).exists());
-    }
+/// Given our saved usage data and our last seen value, process the vecdeques so that we only
+/// send to ops new data since last seen.
+pub fn process_usage_data(
+    mut usage_data_client: VecDeque<RCUsageHour>,
+    mut usage_data_relay: VecDeque<RCUsageHour>,
+    last_seen_hour: u64,
+    current_hour: u64,
+) -> Option<UsageTracker> {
+    // sort client and relay data in case they have come out of order somehow. This sorts by index increasing so newest data at back
+    usage_data_relay
+        .make_contiguous()
+        .sort_by(|a, b| a.index.cmp(&b.index));
+    usage_data_client
+        .make_contiguous()
+        .sort_by(|a, b| a.index.cmp(&b.index));
+
+    // so this spits out the index for where last seen is, or the index of the next highest hour(returned in an error).
+    // we take the result -1 just in case, limit 0, since it's possible we might get back an index out of bounds at the back.
```

Comment:
this means we may send a duplicate usage hour to ops right? Because we send result -1. Have you checked if the code there will handle that gracefully?

Reply:
Ops does currently handle in case it has been sent duplicates, but regardless I will need to update ops because as is now I've switched the order in which the data comes in